### DISCUSSION TITLE
Add fields to documents

### DISCUSSION THREAD

**Discussion by valentin-** [pimcore/pimcore#8839] [General]
Hi, 

are there any examples how to add additional fields to documents or extend the default page document (new page type) and add fields? I'd like to add fields to the settings tab or create an additional tab including those fields. I know this could get achieved through properties, but the values should be mandatory and adding properties is to inconvenient in this case imho. 

Thanks!

Pimcore version: dev-master|add74462508e662a032f297149e1a7dfe5c042d2

**Comment by ckemptner** [comment]
What would you like to achieve? =) What will these fields be used for?

**Reply by valentin-** [reply]
Project detail pages area created using a custom document type. The project overview (areabrick) fetches those pages and creates a list of projects showing the main image, teaser text and category (data object).

I'd like to assign those fields directly in the main settings or a separate tab because they are mandatory - this would be more handy as assigning predefined properties.

In other words: I want to ensure all mandatory values are set correctly before saving the page.

Thanks!

**Reply by ckemptner** [reply]
Hello @valentin-, this would be a pretty typical use case for "data objects" > Have you considered this yet? You can also combine documents + data objects (configuration object for basic mandatory data + fully flexible document for optional layout data)!

**Reply by valentin-** [reply]
Hi @ckemptner, thanks for your advice, this was my first approach. But then i switched to documents + predefined properties because i thought it would be more intuitive. At the end I wasn't happy with this approach either and thought additional fields in the project documents would fit the best. But I guess this is to complicated and I'll move back to documents + data objects.
